public/app/plugins/datasource/influxdb/components/editor/config-v2/AuthSettings.tsx

```diff
@@ -54,36 +61,38 @@ export const AuthSettings = (props: Props) => {
     [options, onOptionsChange]
   );
 
-  /**
-   * Selected authentication method. Fallback to the most common if the selected one is missing.
-   */
   const isAuthMethod = (v: unknown): v is AuthMethod =>
     v === AuthMethod.NoAuth || v === AuthMethod.BasicAuth || v === AuthMethod.OAuthForward;
 
-  const selectedMethod = useMemo<AuthMethod | undefined>(() => {
-    if (isAuthMethod(authProps.selectedMethod)) {
-      return authProps.selectedMethod;
-    }
-    return isAuthMethod(authProps.mostCommonMethod) ? authProps.mostCommonMethod : undefined;
-  }, [authProps.selectedMethod, authProps.mostCommonMethod]);
-
-  /**
-   * Local UI state
-   */
   const [authOptions, setAuthOptions] = useState<AuthOptionState>({
-    basicAuth: selectedMethod === AuthMethod.BasicAuth,
+    noAuth: (!options.basicAuth && !options.jsonData.oauthPassThru) ?? false,
+    basicAuth: options.basicAuth ?? false,
     tlsClientAuth: authProps.TLS?.TLSClientAuth.enabled ?? false,
     caCert: authProps.TLS?.selfSignedCertificate.enabled ?? false,
     skipTLS: authProps.TLS?.skipTLSVerification.enabled ?? false,
-    oAuthForward: selectedMethod === AuthMethod.OAuthForward,
+    oAuthForward: options.jsonData.oauthPassThru ?? false,
     withCredentials: options.withCredentials ?? false,
   });
 
-  /**
-   * Expand/collapse top–level section
-   */
-  const [authenticationSettingsIsOpen, setAuthenticationSettingsIsOpen] = useState(
-    Object.values(authOptions).some(Boolean)
+  const selectedMethod = useMemo<AuthMethod | undefined>(() => {
+    if (isAuthMethod(authProps.selectedMethod) && authProps.selectedMethod !== AuthMethod.CrossSiteCredentials) {
+      return authProps.selectedMethod;
+    }
+
+    switch (true) {
```

Comment:
Is this intentional? I think it might be better to make the condition of the `switch` validate if `authOptions` is defined instead. Wdyt?

Reply:
agreed. I've updated this.